app/src/main/java/com/isc/hermes/model/Utils/IncidentsUtils.java

```diff
@@ -0,0 +1,79 @@
+package com.isc.hermes.model.Utils;
+
+import com.isc.hermes.database.IncidentsUploader;
+
+import org.bson.types.ObjectId;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This is a Utility class for handling incidents-related operations.
+ */
+public class IncidentsUtils {
+    private static IncidentsUtils instance;
+    /**
+     * This method generates a new ObjectId as a hexadecimal string.
+     *
+     * @return The generated ObjectId as a hexadecimal string.
+     */
+    public String generateObjectId() {
+        ObjectId objectId = new ObjectId();
+        return objectId.toHexString();
+    }
+    /**
+     * This method adds a specified time duration to the current date and returns it as a formatted string.
+     *
+     * @param timeDuration The time duration to add in the format "X unit" (e.g., "2 hours", "1 day").
+     * @return The modified date as a formatted string.
+     */
+    public String addTimeToCurrentDate(String timeDuration) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(new Date());
+        Map<String, Integer> unitMap = new HashMap<>();
+        unitMap.put("minute", Calendar.MINUTE);
+        unitMap.put("minutes", Calendar.MINUTE);
+        unitMap.put("hour", Calendar.HOUR);
+        unitMap.put("hours", Calendar.HOUR);
+        unitMap.put("day", Calendar.DAY_OF_YEAR);
+        unitMap.put("days", Calendar.DAY_OF_YEAR);
+        unitMap.put("month", Calendar.MONTH);
+        unitMap.put("months", Calendar.MONTH);
```

Comment:
You could move this unitMap as a class attribute and add the keys and values in the constructor.

Reply:
This observation has been resolved.